### PR TITLE
Refactor: Remove usage of imp

### PR DESCRIPTION
**Pull request by farisachugthai**
Since there are no reasonable alternatives to imp in python2.7,
I figured it'd be the easiest to try importing python3 modules and if
that fails, fall back to the previous method. However this will remove
that annoying DeprecationWarning for everyone using python3.

### CODE REVIEW

**Comment by bfredl** [comment]
don't be too afraid to break 2.7 if there is clear benefit. pynvim 0.5 can be py3.5+ only if we want to.

**Comment by farisachugthai** [comment]
Hahaha thanks for the tip! I wanted this refactored to a separate function so that if the decision to go pure py3 gets made we have the weird py2 / py3 thing contained to 1 function and can simply remove the `except ImportError: import imp` malarky

**Comment by Shougo** [comment]
I think it fixes my issue.
https://github.com/neovim/pynvim/issues/478

The PR is completed?

**Comment by Shougo** [comment]
I think `compat.py` and `scripts_host.py` needs to be changed.

**Comment by Shougo** [comment]
I have imported your commit in https://github.com/neovim/pynvim/pull/486.

File: pynvim/plugin/host.py [hunk 1]
```diff
@@ -23,6 +22,27 @@
...
+    try:  # Python3
+        from importlib.util import module_from_spec, spec_from_file_location
+    except ImportError:  # Python2.7
+        import imp
+        from pynvim.compat import find_module
+        file, pathname, descr = find_module(name, [directory])
+        module = imp.load_module(name, file, pathname, descr)
+        return module
+    else:
+        spec = spec_from_file_location(name, location=directory)
```
Comment (wookayin) [inline pynvim/plugin/host.py:39]: I don't think this is correct -- this seems to be always returning `None`, causing `ImportError`. `
```
spec_from_file_location("foobar", "/path/to/foobar.py") == ModuleSpec(name='foobar', ...)
spec_from_file_location("foobar", "/path/to/") == None
```

File: pynvim/plugin/host.py [hunk 2]
```diff
@@ -161,8 +181,10 @@ def _load(self, plugins):
                     has_script = True
                 else:
                     directory, name = os.path.split(os.path.splitext(path)[0])
-                    file, pathname, descr = find_module(name, [directory])
-                    module = imp.load_module(name, file, pathname, descr)
+                    try:
+                        module = handle_import(directory, name)
+                    except ImportError:
+                        return
```
Comment (wookayin) [inline pynvim/plugin/host.py:187]: pynvim remote plugin stopped working with handler/autocmd not found errors, but it was being silently ignored here

Comment (justinmk) [reply pynvim/plugin/host.py:187]: Thanks for noticing that. Good thing we didn't release yet :D Any chance you can send a fix?

Comment (wookayin) [reply pynvim/plugin/host.py:187]: I managed to fix this. I'm sending a PR very soon.

**Pull request merged**